### PR TITLE
Automated cherry pick of #120550: DRA: call plugins for claims even if exist in cache

### PR DESCRIPTION
**Pull request by adrianchiris**
Cherry pick of #120550 on release-1.28.

#120550: DRA: call plugins for claims even if exist in cache

For details on the cherry pick process, see the [cherry pick requests](https://git.k8s.io/community/contributors/devel/sig-release/cherry-picks.md) page.

```release-note

```

### CODE REVIEW

**Comment by adrianchiris** [comment]
@bart0sh @klueska PTAL

going over [cherry-picks](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-release/cherry-picks.md) i see that alpha feature related bugs are not considered critical, so if this does not qualify for backport feel free to close this one.

Reason to add it:
- This fixes an important (IMO) issue with DRA so we would like to have this fix as part of a released k8s version
- the fix is isolated to DRA so risk is low.

**Comment by bart0sh** [comment]
/assign @klueska @mrunalp
for a SIG-node approval

**Comment by adrianchiris** [comment]
@bart0sh is there anything else required ?

**Comment by bart0sh** [comment]
@adrianchiris We're waiting for a [Release managers](https://github.com/orgs/kubernetes/teams/release-managers) approval as far as I understand.

**Comment by adrianchiris** [comment]
> How big is the risk that we introduce a regression here? It feels that we changed a non-trivial behavior, right?

Adding my POV:
I think the risk is acceptable as without this change, after reboot, pod will not have dra resources.
the change is isolated to dra manager in kubelet. so will not affect other flows.

**the gist of the change:**
instead of calling dra plugin once per resource (checkpointed, never called again),
we call dra plugin once per resource on kubelet start and only if a pod which requests dra resources need to be prepared.

note: dra plugins are defined as idempotent, so callling more than once with same parameters should yield same result.

**Review by saschagrunert — APPROVED** [review]

**Pull request merged**
